### ISSUE TITLE
Easier hyperparameter tuning?

### ISSUE THREAD

**Issue by nabsiddiqui** [selcukorkmaz/fastml#3]
This is a great package and desperately needed in R. Python has Lazy Predict which has been useful for me in the past but I don’t like having to switch to Python for it. 

I was wondering if there is a plan to make it easier to do hyperparameter tuning? Maybe incorporating something like EZtune: https://cran.r-project.org/web/packages/EZtune/vignettes/my-vignette.html

**Comment by selcukorkmaz** [comment]
This is something we definitely consider to add to our package. Thank you for pointing it out and thank you for the valuable feedback.

**Comment by nabsiddiqui** [comment]
@selcukorkmaz Perhaps one way to start is to implement nested cross-validation? Currently, the k-fold cross-validation averages the scores for each fold (if I understand your code correctly), but since you are technically using the training data for training, I believe you should still use a validation set or nested cross-validation. The latter has the benefit of allowing you to later expand the package for hyperparameter turning.
